static/app/components/tabs/tabList.tsx

```diff
@@ -80,11 +81,21 @@ function BaseTabList({hideBorder = false, className, ...props}: TabListProps) {
   const ariaProps = {
     selectedKey: value,
     defaultSelectedKey: defaultValue,
-    onSelectionChange: onChange,
+    onSelectionChange: key => {
+      onChange?.(key);
+
+      // If the newly selected tab is a tab link, then navigate to the specified link
+      const linkTo = [...(props.items ?? [])].find(item => item.key === key)?.to;
+      if (!linkTo) {
+        return;
+      }
+      browserHistory.push(linkTo);
```

Comment:
You shouldn't need to do this, clicking the link already pushes to browser history.

Reply:
This is to account for keyboard navigation. When the tab list is focused, you can use the arrow left/right keys to move between tabs. Without this, pressing an arrow key will only change the tab selection but not the URL.

Reply:
Hmm, maybe this is out of scope for this PR, but it seems to me like we wouldn't want automatic activation on focus for our tabs since they are rendering a lot of content that needs to be fetched and such. Maybe it would be better (and simpler to code) if we required the user to press tab/enter after a tab was focused.

https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#kbd_selection_follows_focus

> However, if displaying a new panel causes a network request and possibly a page refresh, the effect of having selection automatically focus can be devastating to the experience for keyboard and screen reader users. In this case, displaying the fourth tab or reading through the list becomes a tedious and time-consuming task as the user experiences significant latency with each movement of focus.

There's a example of both types on this page: https://www.w3.org/WAI/ARIA/apg/patterns/tabpanel/

Reply:
Oh I can do that by changing one of the props passed to `useTabList`!

Reply:
Done! https://github.com/getsentry/sentry/pull/40852/commits/4b115db78ac65c86ba5e2814e69997e87a01cc18

Reply:
It doesn't remove the need for a `browserHistory` push though - pressing enter/space still won't work without it.

Reply:
Fair enough! I'm happy that it was easy to change to manual activation 😄